cloud/blockstore/libs/notify/notify.cpp

```diff
@@ -152,11 +158,31 @@ class TService final
             data.Event);
 
         auto p = NewPromise<NProto::TError>();
+        
+        std::optional<TString> iamTokenStr;
+        if (Config->GetVersion() == 2){
+            if (!IamClient) {
+                STORAGE_WARN("missing iam-client "
+                    <<"Got error while requesting token: " <<
+                        "IAM client is missing");
+            } else {
+                try {
+                    auto tokenInfo = IamClient->GetTokenAsync().GetValue(WaitTimeout);
```

Comment:
It may blocks the current thread for 100ms. Doesn't look great.
Why not use a continuation? Something like this:
```cpp

GetIAMToken().Subscribe([this, p, event = data.Event] (TString token) {
    HttpsClient.Post(
        Config->GetEndpoint(),
        v.GetStringRobust(),
        "application/json",
        std::move(token),
        [p, event = data.Event] (int code, const TString& message) mutable {
            ...
        });
});

auto GetIAMToken()
{
    if (Config->GetVersion() == 2) {
        // ...

        return IamClient->GetTokenAsync().Apply([] (const auto& future) {
            return future.GetValue().GetResult().Token;
        });
    }

    return MakeFuture(TString{});
}
```